Added function to return Model Id

ChuckVanzant opened the pull request:
Added function to return the word from register Model_ID (0x010F) per the datasheet.

Note that my Adafruit **VL53L4CX** Time of Flight Distance Sensor returns **0xEBAA** for the word at index **0x010F**, the same as the **VL53L4CD** sensor, and not the value of **0xEAAA** documented in the **VL53L4CX** datasheet (3.2 I2C interface - reference registers).

Could Adafruit have happened to integrate the VL53L4CD sensor into their “VL53L4CX” device? I’ll have to inquire with Adafruit.

ChuckVanzant committed:
Added function to return Model Id

Added function to return Model Id from register Model_ID (0x010F) per the datasheet.

Note that my Adafruit VL53L4CX Time of Flight Distance Sensor returns 0xEBAA for the word at index 0x010F, the same as the VL53L4CD sensor, and not the value of 0xEAAA documented in the VL53L4CX datasheet (3.2 I2C interface - reference registers).

ChuckVanzant committed:
Merge branch 'stm32duino:main' into main

cparata commented:
Hello @ChuckVanzant ,
actually the output that you read is compatible with the Datasheet of VL53L4CX.

![DS_VL53L4CX](https://user-images.githubusercontent.com/16096427/177145279-08c1436d-a716-40d3-8093-f9c3557adc6d.PNG)

Best Regards,
Carlo

cparata commented:
Hi @ChuckVanzant ,
the register address 0x010F is already defined in vl53l4cx_register_map.h (VL53L4CX_IDENTIFICATION__MODEL_ID). Could you update the PR, using this define?
Thanks in advance and Best Regards,
Carlo

cparata commented on src/vl53l4cx_def.h:
```diff
@@ -31,7 +31,8 @@ extern "C" {
 
-
+/** VL53L4CX IMPLEMENTATION model id */
+#define VL53L4CX_IDENTIFICATION_MODEL_ID       ((uint16_t)0x010F)
```
This define is useless, because you can use the define "VL53L4CX_IDENTIFICATION__MODEL_ID" that you can find in vl53l4cx_register_map.h.

ChuckVanzant replied:
Thanks for all your responses and clarifications!

I must have been referencing an older revision of the datasheet.

Based on my previous assumption, I was assuming the Sensor ID of the device (Model_ID and Module_Type) would uniquely identify which sensor is in use.

Since the VL53L4CX and VL53L4CD sensors have the same Model_ID and Module_Type, is there a way to query the device to determine which sensor it is?

I will update the PR. I will also change the name of the proposed function from VL53L4CX_GetModelId to VL53L4CX_GetSensorId, akin to the function VL53L4CD_GetSensorId for the VL53L4CD sensor.

cparata replied:
Hello @ChuckVanzant ,
maybe you can use the firmware version to distinguish between the 2 devices. VL53L4CD uses firmware version 1.0.0 and VL53L4CX uses 1.2.8.
But, it is just an idea.
Best Regards,
Carlo

ChuckVanzant committed:
Change Function Name, Use Existing Register Define

The name of the proposed function has been changed from VL53L4CX_GetModelId to VL53L4CX_GetSensorId, akin to the function VL53L4CD_GetSensorId for the VL53L4CD sensor.

The function was also changed to use an existing register define and the unnecessary define originally submitted has been removed.
